test/evolve.js

```diff
@@ -11,18 +11,26 @@ describe('evolve', function() {
     assert.deepEqual(R.evolve(transf, object), expected);
   });
 
-  it('invokes function with undefined if object does not contain the key', function() {
-    var transf   = {wow: String};
-    var object   = {};
-    var expected = {wow: 'undefined'};
+  it('does not invoke function if object does not contain the key', function() {
+    var transf   = {n: R.add(1), m: R.add(1)};
+    var object   = {m: 3};
+    var expected = {m: 4};
     assert.deepEqual(R.evolve(transf, object), expected);
   });
 
   it('is not destructive', function() {
     var transf   = {elapsed: R.add(1), remaining: R.add(-1)};
     var object   = {name: 'Tomato', elapsed: 100, remaining: 1400};
+    var expected = {name: 'Tomato', elapsed: 100, remaining: 1400};
     R.evolve(transf, object);
-    assert.deepEqual(object, {name: 'Tomato', elapsed: 100, remaining: 1400});
+    assert.deepEqual(object, expected);
+  });
+
+  it('is recursive', function() {
+    var transf   = {nested: {second: R.add(-1), third: R.add(1)}};
+    var object   = {first: 1, nested: {second: 2, third: 3}};
+    var expected = {first: 1, nested: {second: 1, third: 4}};
+    assert.deepEqual(R.evolve(transf, object), expected);
   });
```

Comment:
Ugh.  JS strikes again!

Obviously the implementation will do so.  But is it really what we want?  So we have to then check that the object and transformation both have an array, and both have the same length?  Meh!